superset/assets/javascripts/explore/actions/chartActions.js

```diff
@@ -56,19 +102,23 @@ export function runQuery(formData, force = false, timeout = 60) {
     const queryRequest = $.ajax({
       url,
       dataType: 'json',
-      success(queryResponse) {
-        dispatch(chartUpdateSucceeded(queryResponse));
-      },
-      error(err) {
+      timeout: timeout * 1000,
+    });
+    const queryPromise = queryRequest
```

Comment:
I changes this to use promises to make everything composeable. I tested this a lot....
https://github.com/gaearon/redux-thunk#composition